testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergProcedureCalls.java

```diff
@@ -59,6 +60,103 @@ public void testMigrateHiveTable()
+    {
+        String tableName = "test_migrate_timestamp" + randomNameSuffix();
+        String hiveTableName = "hive.default." + tableName;
+
+        onTrino().executeQuery("DROP TABLE IF EXISTS " + hiveTableName);
+        onTrino().executeQuery("CREATE TABLE " + hiveTableName + " (x timestamp(3)) WITH (format='AVRO')");
+        onTrino().executeQuery("INSERT INTO " + hiveTableName + " VALUES timestamp '2021-01-01 10:11:12.123'");
+
+        assertThatThrownBy(() -> onTrino().executeQuery("CALL iceberg.system.migrate('default', '" + tableName + "')"))
+                .hasMessageContaining("Migrating timestamp type with AVRO format is not supported.");
+
+        onTrino().executeQuery("DROP TABLE IF EXISTS " + hiveTableName);
+    }
+
+    @Test(groups = {ICEBERG, PROFILE_SPECIFIC_TESTS})
+    public void testMigrateTimestampHiveTableWithParquet()
+    {
+        asList(
+                new TimestampTestCase("MILLISECONDS",
+                        3, "2021-01-01 10:11:12.123",
+                        "2021-01-01 10:11:12.123000 UTC",
+                        "2021-01-01 10:11:12.123"),
+                new TimestampTestCase("MICROSECONDS",
+                        6, "2021-01-01 10:11:12.123456",
+                        "2021-01-01 10:11:12.123456 UTC",
+                        "2021-01-01 10:11:12.123456"),
+                new TimestampTestCase("NANOSECONDS",
+                        9, "2021-01-01 10:11:12.123456789",
+                        "2021-01-01 10:11:12.123457 UTC",
+                        "2021-01-01 10:11:12.123456")
+        ).forEach(testCase ->
+                testMigrateTimestampHiveTable(
+                        testCase.precisionName,
+                        testCase.precision,
+                        testCase.inputValue,
+                        testCase.expectedValueInTrino,
+                        testCase.expectedValueInSpark,
+                        "PARQUET"));
+    }
+
+    private record TimestampTestCase(String precisionName, int precision, String inputValue, String expectedValueInTrino, String expectedValueInSpark) {}
+
+    private void testMigrateTimestampHiveTable(String precisionName, int precision, String inputValue, String expectedValueInTrino, String expectedValueInSpark, String format)
+    {
+        String tableName = "test_migrate_timestampPar_" + randomNameSuffix();
+        String hiveTableName = "hive.default." + tableName;
+        String icebergTableName = "iceberg.default." + tableName;
+        String sparkTableName = "iceberg_test.default." + tableName;
+
+        onTrino().executeQuery("DROP TABLE IF EXISTS default." + tableName);
+        onTrino().executeQuery("SET SESSION hive.timestamp_precision = '" + precisionName + "'");
+        onTrino().executeQuery("CREATE TABLE " + hiveTableName + " (x timestamp(" + precision + ")) WITH (format='" + format + "')");
+        onTrino().executeQuery("INSERT INTO " + hiveTableName + " VALUES timestamp '" + inputValue + "'");
```

Comment:
Can we remove `precision` variable if we use `CREATE TABLE AS SELECT` statement?